lib/Parse/ParseType.cpp

```diff
@@ -1474,7 +1474,9 @@ static bool isGenericTypeDisambiguatingToken(Parser &P) {
   auto &tok = P.Tok;
   switch (tok.getKind()) {
   default:
-    return false;
+    // If this is the end of the expr (wouldn't match parseExprSequenceElement),
+    // prefer generic type list over an illegal unary postfix '>' operator.
+    return tok.isKeyword() && tok.getKind() != tok::kw_try;
```

Comment:
@rintaro could you take a look at this please?

Reply:
`tok.isKeyword()` is usually not enough for detecting "the token is not related to the current production" because many decls starts with a non-keyword.  e.g.
```swift
let a = Foo<Bar>
open class Cls { ... }
```

`P.isStartOfSwiftDecl() || P.isStartOfStmt()` might be better.